src/demo/navbar/navbar-top.component.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  ViewEncapsulation
+} from '@angular/core';
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'navbar-top',
+  styleUrls: ['./navbar-top.component.less'],
+  templateUrl: './navbar-top.component.html'
+})
+export class NavbarTopComponent implements OnInit {
+  @Input() navItem: string;
+
+  @Output('onNavItemSelect') onNavItemSelect = new EventEmitter();
+  @Output('onNavbarToggle') onNavbarToggle = new EventEmitter();
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+  }
+
+  isNavItemSelected(name: string): boolean {
+    return this.navItem === name;
+  }
+
+  selectNavItem($event: MouseEvent, name: string): void {
+    this.navItem = name;
```

Comment:
Is it typical to assign a value to an Input in angular? Seems odd to me.

Reply:
tbh I've taken example from https://github.com/patternfly/patternfly-ng/tree/master/src/demo/navbar (I really think this is a great demo) i think assigning the @input here means: refresh the view.